llmfoundry/data/finetuning/dataloader.py

```diff
@@ -145,7 +148,31 @@ def build_finetuning_dataloader(cfg: DictConfig, tokenizer: Tokenizer,
         )
 
     else:
-        dataset = dataset_constructor.build_from_hf(cfg.dataset, tokenizer)
+        backend, _, _ = parse_uri(cfg.dataset.hf_name)
+        if backend not in ['', None]:
+            if cfg.dataset.get('split') is None:
+                raise ValueError(
+                    'When using a HuggingFace dataset from a URL, you must set the ' +\
+                    '`split` key in the dataset config.'
+                )
+            with tempfile.TemporaryDirectory() as tmp_dir:
+                supported_extensions = ['jsonl', 'csv', 'parquet']
+                for extension in supported_extensions:
+                    name = f'{cfg.dataset.hf_name}/{cfg.dataset.split}.{extension}'
+                    location = f'{tmp_dir}/{name}'
+                    try:
+                        get_file(cfg.dataset.hf_name, location)
```

Comment:
you probably want to only do the downloading on rank 0 (and make everyone else wait for the download to be finished

Reply:
gee, if only there were a nice utility for that

Reply:
@dakinggg sorry, fixed